flang/module/__cuda_device_builtins.f90

```diff
@@ -0,0 +1,74 @@
+!===-- module/__cuda_device_builtins.f90 -----------------------------------===!
```

Comment:
These could all be in `__fortran_builtins`, which already contains some built-in CUDA definitions.

Reply:
The reason I added a new file is that we can check on the module name instead of having to check on the module name and the procedure prefix. 
I can moved them in __fortran_builtins but we would need to check if the procedure names starts with `__cuda_device_builtins_` as well.